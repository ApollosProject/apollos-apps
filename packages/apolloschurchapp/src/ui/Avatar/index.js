import React from 'react';
import PropTypes from 'prop-types';
import { View } from 'react-native';
import { get } from 'lodash';
import { compose, pure } from 'recompose';

import ConnectedImage from 'apolloschurchapp/src/ui/ConnectedImage';
import styled from 'apolloschurchapp/src/ui/styled';
import { withTheme } from 'apolloschurchapp/src/ui/theme';
import ActivityIndicator from 'apolloschurchapp/src/ui/ActivityIndicator';

export { default as AvatarList } from './List';

const enhance = compose(
  pure,
  withTheme(({ theme, size }) => ({
    themeSize: get(theme.sizing.avatar, size, theme.sizing.avatar.small),
  }))
);

const Container = styled(
  ({ theme, themeSize }) => ({
    width: themeSize,
    height: themeSize,
    backgroundColor: theme.colors.background.inactive,
    borderRadius: themeSize / 2,
    marginRight: themeSize / 4,
    marginBottom: themeSize / 4,
    overflow: 'hidden',
  }),
  'Avatar'
)(View);

const LoadingIcon = compose(
  withTheme(({ theme: { colors } = {} }) => ({ color: colors.white })),
  styled({
    zIndex: 1,
  })
)(ActivityIndicator);

const Image = styled(({ themeSize }) => ({
  width: '100%',
  height: '100%',
  position: 'absolute',
  top: 0,
  left: 0,
  right: 0,
  bottom: 0,
  borderRadius: themeSize / 2,
}))(ConnectedImage);

const Avatar = enhance(
  ({ themeSize, containerStyle, source, isLoading, ...imageProps }) => (
    <Container style={containerStyle} themeSize={themeSize}>
      {isLoading ? <LoadingIcon /> : null}
      {source && source.uri ? (
        <Image
          source={source}
          {...imageProps}
          themeSize={themeSize}
          isLoading={isLoading}
        />
      ) : null}
    </Container>
  )
);

Avatar.propTypes = {
  size: PropTypes.oneOf(['small', 'medium', 'large']),
  containerStyle: PropTypes.any, // eslint-disable-line
  ...ConnectedImage.propTypes,
};

export default Avatar;
