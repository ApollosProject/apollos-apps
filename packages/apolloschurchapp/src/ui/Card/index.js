export { default } from './CardWrapper';
export CardImage from './Image';
export CardContent from './Content';
export CardActions from './Actions';
export ErrorCard from './ErrorCard';
