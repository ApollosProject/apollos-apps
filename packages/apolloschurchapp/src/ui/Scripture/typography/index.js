import { Text } from 'react-native';

import styled from 'apolloschurchapp/src/ui/styled';

import ScriptureText from './ScriptureText';

const VerseNumber = styled(({ theme }) => ({
  fontSize: theme.helpers.rem(0.6),
  color: theme.colors.text.secondary,
}))(ScriptureText);

const RedLetters = styled(({ theme }) => ({
  color: theme.colors.wordOfChrist,
}))(Text);

const PoeticPause = styled({
  textAlign: 'right',
})(ScriptureText);

export { ScriptureText, VerseNumber, RedLetters, PoeticPause };
