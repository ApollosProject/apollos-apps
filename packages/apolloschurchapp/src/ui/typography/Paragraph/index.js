import { compose, setDisplayName } from 'recompose';
import PropTypes from 'prop-types';
import { View } from 'react-native';

import {
  withPlaceholder,
  Paragraph as ParagraphPlaceholder,
} from 'apolloschurchapp/src/ui/Placeholder';
import styled from 'apolloschurchapp/src/ui/styled';

const Paragraph = compose(
  setDisplayName('Paragraph'),
  styled(({ theme }) => ({
    paddingBottom: theme.helpers.verticalRhythm(0.75),
  })),
  withPlaceholder(ParagraphPlaceholder)
)(View);

Paragraph.propTypes = {
  isLoading: PropTypes.bool, // display loading placeholder
};

export default Paragraph;
