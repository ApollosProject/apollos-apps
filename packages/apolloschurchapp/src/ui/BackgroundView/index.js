import LinearGradient from 'react-native-linear-gradient';
import { compose } from 'recompose';

import styled from 'apolloschurchapp/src/ui/styled';
import { withTheme } from 'apolloschurchapp/src/ui/theme';

const BackgroundView = compose(
  withTheme(({ theme }) => ({
    colors: [theme.colors.background.paper, theme.colors.background.screen],
  })),
  styled({ flex: 1 })
)(LinearGradient);

export default BackgroundView;
