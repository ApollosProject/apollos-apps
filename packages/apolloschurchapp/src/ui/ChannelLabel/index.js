import React from 'react';
import { View } from 'react-native';
import PropTypes from 'prop-types';
import { compose, pure } from 'recompose';
import { kebabCase } from 'lodash';

import { withIsLoading } from 'apolloschurchapp/src/ui/isLoading';
import { withTheme } from 'apolloschurchapp/src/ui/theme';
import styled from 'apolloschurchapp/src/ui/styled';
import { H6 } from 'apolloschurchapp/src/ui/typography';
import Icon from 'apolloschurchapp/src/ui/Icon';
import * as Icons from 'apolloschurchapp/src/ui/Icon/icons';

const enhance = compose(
  withIsLoading,
  pure,
  withTheme()
);

const Wrapper = styled(({ flexed }) => ({
  flex: flexed ? 1 : null,
  flexDirection: 'row',
  alignItems: 'center',
}))(View);

const PlaceholderWrapper = styled(({ theme, withIcon }) => ({
  flex: 1,
  ...(withIcon
    ? { paddingHorizontal: theme.sizing.baseUnit / 2 }
    : { paddingRight: theme.sizing.baseUnit / 2 }),
}))(View);

const StyledH6 = styled(({ theme }) => ({
  color: theme.colors.text.secondary,
}))(H6);

const ChannelLabel = enhance(({ label, icon, withFlex, isLoading, theme }) => (
  <Wrapper flexed={withFlex}>
    {icon ? (
      <Icon
        name={icon}
        size={theme.helpers.rem(1.2)}
        fill={theme.colors.text.secondary}
        isLoading={isLoading}
      />
    ) : null}
    <PlaceholderWrapper withIcon={icon}>
      <StyledH6>{label}</StyledH6>
    </PlaceholderWrapper>
  </Wrapper>
));

ChannelLabel.propTypes = {
  label: PropTypes.oneOfType([PropTypes.string, PropTypes.element]).isRequired,
  icon: PropTypes.oneOf(Object.keys(Icons).map(kebabCase)),
  isLoading: PropTypes.bool,
  withFlex: PropTypes.bool,
};

ChannelLabel.defaultProps = {
  withFlex: false,
};

export default ChannelLabel;
