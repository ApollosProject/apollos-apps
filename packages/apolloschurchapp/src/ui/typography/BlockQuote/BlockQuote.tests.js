import React from 'react';
import renderer from 'react-test-renderer';

import Providers from 'apolloschurchapp/src/Providers';
import { H5 } from 'apolloschurchapp/src/ui/typography';
import { withIsLoading } from 'apolloschurchapp/src/ui/isLoading';

import BlockQuote from '.';

describe('the BlockQuote component', () => {
  it('should render a string', () => {
    const tree = renderer.create(
      <Providers>
        <BlockQuote>
          {
            '"You are the only Bible some unbelievers will ever read." – John MacArthur'
          }
        </BlockQuote>
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render child components', () => {
    const tree = renderer.create(
      <Providers>
        <BlockQuote>
          <H5>
            {
              '"You are the only Bible some unbelievers will ever read." – John MacArthur'
            }
          </H5>
        </BlockQuote>
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render a loading state', () => {
    const BlockQuoteLoadingState = withIsLoading(BlockQuote);
    const tree = renderer.create(
      <Providers>
        <BlockQuoteLoadingState isLoading>
          {
            '"You are the only Bible some unbelievers will ever read." – John MacArthur'
          }
        </BlockQuoteLoadingState>
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
});
