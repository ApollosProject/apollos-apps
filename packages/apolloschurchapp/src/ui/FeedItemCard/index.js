import React from 'react';
import PropTypes from 'prop-types';
import { compose, pure } from 'recompose';
import { startCase, toLower } from 'lodash';

import { withThemeMixin, withTheme } from 'apolloschurchapp/src/ui/theme';
import Icon from 'apolloschurchapp/src/ui/Icon';
import ChannelLabel from 'apolloschurchapp/src/ui/ChannelLabel';
import GradientOverlayImage from 'apolloschurchapp/src/ui/GradientOverlayImage';
import Card, { CardContent, CardActions } from 'apolloschurchapp/src/ui/Card';
import { H3 } from 'apolloschurchapp/src/ui/typography';
// import ConnectedImage from 'apolloschurchapp/src/ui/ConnectedImage';
// import ProgressiveImage from 'apolloschurchapp/src/ui/ProgressiveImage';

// import LikeButton from './LikeButton';

const enhance = compose(
  pure,
  withThemeMixin(({ isLight }) => ({
    type: isLight ? 'light' : 'dark',
  })),
  withTheme()
);

const FeedItemCard = enhance(
  ({
    images,
    title,
    channelType,
    channelTypeIcon,
    isLoading,
    isLight,
    isLiked,
    backgroundColor,
    theme,
    id,
    ...otherProps
  }) => (
    <Card isLoading={isLoading} cardColor={backgroundColor} {...otherProps}>
      <GradientOverlayImage
        isLoading={isLoading}
        source={images}
        overlayColor={backgroundColor}
      />
      <CardContent>
        <H3 isLoading={isLoading}>{title}</H3>
      </CardContent>
      <CardActions>
        <ChannelLabel
          label={startCase(toLower(channelType))}
          icon={channelTypeIcon}
          isLoading={isLoading}
          withFlex
        />
        <Icon
          name={isLiked ? 'like-solid' : 'like'}
          size={theme.helpers.rem(1.2)}
          fill={theme.colors.text.primary}
          isLoading={isLoading}
        />
      </CardActions>
    </Card>
  )
);

const sourcePropType = PropTypes.oneOfType([
  PropTypes.arrayOf(
    PropTypes.shape({
      uri: PropTypes.string,
    })
  ),
  PropTypes.string,
]);

FeedItemCard.propTypes = {
  title: PropTypes.string.isRequired,
  images: sourcePropType,
  channelType: PropTypes.string,
  channelTypeIcon: PropTypes.string,
  isLoading: PropTypes.bool,
  isLiked: PropTypes.bool,
  isLight: PropTypes.bool,
  backgroundColor: PropTypes.string,
  style: PropTypes.any, // eslint-disable-line
};

FeedItemCard.defaultProps = {
  isLight: true,
  channelTypeIcon: null,
};

export default FeedItemCard;
