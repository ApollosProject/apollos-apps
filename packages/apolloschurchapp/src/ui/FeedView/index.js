import React, { Component } from 'react';
import { FlatList, TouchableWithoutFeedback } from 'react-native';
import PropTypes from 'prop-types';
import { pure, compose, branch, withProps, defaultProps } from 'recompose';
import { get } from 'lodash';

import FeedItemCard from 'apolloschurchapp/src/ui/FeedItemCard';
import { enhancer as mediaQuery } from 'apolloschurchapp/src/ui/MediaQuery';
import { withTheme } from 'apolloschurchapp/src/ui/theme';
import { ErrorCard } from 'apolloschurchapp/src/ui/Card';

const StyledFlatList = compose(
  withTheme(({ theme: { helpers: { verticalRhythm } } } = {}) => ({
    contentContainerStyle: { paddingVertical: verticalRhythm(0.3125) }, // Android padding fix ðŸ˜©
  }))
)(FlatList);

class FeedView extends Component {
  static propTypes = {
    content: PropTypes.array, // eslint-disable-line
    error: PropTypes.oneOfType([
      PropTypes.bool,
      PropTypes.string,
      PropTypes.object,
    ]),
    fetchMore: PropTypes.func,
    isLoading: PropTypes.bool,
    keyExtractor: PropTypes.func,
    ListEmptyComponent: PropTypes.func,
    numColumns: PropTypes.number,
    onEndReachedThreshold: PropTypes.number,
    onPressItem: PropTypes.func,
    refetch: PropTypes.func,
    renderEmptyState: PropTypes.func,
    renderItem: PropTypes.func,
  };

  static defaultProps = {
    isLoading: false,
    // renderItem: this.defaultFeedItemRenderer,
    onEndReachedThreshold: 0.7,
    keyExtractor: (item) => item && item.id,
    content: [],
    refetch: undefined,
    fetchMore: undefined,
  };

  refetchHandler = ({ isLoading, refetch }) =>
    refetch && ((...args) => !isLoading && refetch(...args));

  fetchMoreHandler = ({ fetchMore, error, isLoading }) =>
    fetchMore && ((...args) => !isLoading && !error && fetchMore(...args));

  renderItem = ({ item }) => {
    if (this.props.renderItem) {
      return this.props.renderItem({ item });
    }
    return (
      // These are all props of FeedItemCard but they do not have data coming
      // back yet. So I moved them here for safe keeping.
      // TODO: Move them back when the data is ready.
      // backgroundColor={item.theme.colors.background.paper}
      // isLight={item.theme.isLight}
      // isLiked={item.isLiked}
      <TouchableWithoutFeedback
        onPress={() => this.props.onPressItem({ ...item })}
      >
        <FeedItemCard
          id={get(item, 'id')}
          title={get(item, 'title') || get(item, 'name') || ' '}
          isLiked={get(item, 'isLiked')}
          channelType={get(item, 'parentChannel.name')}
          channelTypeIcon={get(item, 'parentChannel.iconName')}
          images={get(item, 'coverImage.sources')}
          isLoading={get(item, 'isLoading')}
        />
      </TouchableWithoutFeedback>
    );
  };

  render() {
    const {
      content,
      error,
      fetchMore,
      isLoading,
      keyExtractor,
      ListEmptyComponent,
      numColumns,
      onEndReachedThreshold,
      refetch,
      renderEmptyState,
      renderItem,
      ...otherProps
    } = this.props;
    return (
      <StyledFlatList
        {...otherProps}
        data={content}
        keyExtractor={keyExtractor}
        ListEmptyComponent={
          error && !isLoading && (!content || !content.length) ? (
            <ErrorCard error={error} />
          ) : (
            ListEmptyComponent
          )
        }
        numColumns={numColumns}
        onEndReached={this.fetchMoreHandler({ fetchMore, error, isLoading })}
        onEndReachedThreshold={onEndReachedThreshold}
        onRefresh={this.refetchHandler({ isLoading, refetch })}
        refreshing={isLoading}
        renderItem={this.renderItem}
      />
    );
  }
}

const defaultLoadingState = {
  id: 'fakeId0',
  isLoading: true,
  title: '',
  channelType: '',
  coverImage: [],
  theme: {
    isLight: '',
    colors: {
      background: {
        paper: '',
      },
    },
  },
  parentChannel: {
    id: '',
    name: '',
  },
};

const generateLoadingStateData = (
  numberOfItems = 1,
  loadingStateObject = defaultLoadingState
) => {
  const itemData = () => ({ ...loadingStateObject });

  const loadingStateData = [];

  while (loadingStateData.length < numberOfItems) {
    const newData = itemData();
    newData.id = `fakeId${loadingStateData.length}`;
    loadingStateData.push(newData);
  }

  return loadingStateData;
};

const enhance = compose(
  pure,
  branch(
    ({ isLoading, content }) => isLoading && !content.length,
    withProps(({ loadingStateObject }) => ({
      isLoading: true,
      content: generateLoadingStateData(10, loadingStateObject),
      fetchMore: () => {},
    }))
  ),
  mediaQuery(
    ({ md }) => ({ maxWidth: md }),
    defaultProps({ numColumns: 1 }),
    defaultProps({ numColumns: 2 })
  )
);

export default enhance(FeedView);
