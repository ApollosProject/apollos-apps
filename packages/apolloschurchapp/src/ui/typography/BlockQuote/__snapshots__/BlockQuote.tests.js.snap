// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the BlockQuote component should render a loading state 1`] = `
<RCTSafeAreaView
  onLayout={[Function]}
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "stretch",
          "borderBottomWidth": 0.5,
          "borderColor": "#A5A5A5",
          "borderTopWidth": 0.5,
          "marginVertical": 46.08,
          "paddingHorizontal": 16,
          "paddingVertical": 34.56,
        }
      }
    >
      <View
        style={
          Object {
            "alignSelf": "stretch",
            "backgroundColor": "#A5A5A5",
            "borderRadius": 16,
            "height": 16,
            "marginVertical": 5,
            "width": "100%",
          }
        }
      />
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`the BlockQuote component should render a string 1`] = `
<RCTSafeAreaView
  onLayout={[Function]}
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "stretch",
          "borderBottomWidth": 0.5,
          "borderColor": "#A5A5A5",
          "borderTopWidth": 0.5,
          "marginVertical": 46.08,
          "paddingHorizontal": 16,
          "paddingVertical": 34.56,
        }
      }
    >
      <Text
        bold={false}
        italic={false}
        style={
          Object {
            "color": "#303030",
            "fontFamily": "InterUI-Italic",
            "fontSize": 16,
            "lineHeight": 26,
            "textAlign": "center",
          }
        }
      >
        "You are the only Bible some unbelievers will ever read." – John MacArthur
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`the BlockQuote component should render child components 1`] = `
<RCTSafeAreaView
  onLayout={[Function]}
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "stretch",
          "borderBottomWidth": 0.5,
          "borderColor": "#A5A5A5",
          "borderTopWidth": 0.5,
          "marginVertical": 46.08,
          "paddingHorizontal": 16,
          "paddingVertical": 34.56,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#303030",
            "fontFamily": "InterUI-Medium",
            "fontSize": 14,
            "lineHeight": 20.16,
          }
        }
      >
        "You are the only Bible some unbelievers will ever read." – John MacArthur
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
